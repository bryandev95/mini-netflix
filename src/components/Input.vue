<template>
  <div class="input">
    <label v-if="label">{{ label }}</label>
    <input type="text" :name="name" :placeholder="placeholder" :value="value" />
  </div>
</template>

<script>
export default {
  name: "Input",

  props: {
    name: { type: String, required: false, default: "" },
    label: { type: String, required: false, default: "" },
    placeholder: { type: String, required: false, default: "" },
    value: { type: String, required: false, default: "" },
  },
};
</script>

<style scoped lang="scss">
.input {
  display: flex;
  flex-direction: column;

  label {
    color: $gray;
    display: block;
    font-size: $fontSizeDefault;
    font-weight: $fontWeightMedium;
    margin-bottom: 6px;
  }

  input {
    border: 1px solid $gray;
    color: $black;
    font-size: $fontSizeDefault;
    outline: none;
    padding: 8px 12px;
    transition: border 0.2s ease-in-out;

    ::placeholder {
      opacity: 0.4;
    }

    &:focus,
    &:active {
      border-color: $gray;
    }
  }
}
</style>
