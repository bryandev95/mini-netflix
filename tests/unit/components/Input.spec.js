import { shallowMount } from "@vue/test-utils";
import Input from "@/components/Input.vue";

const defaultProps = {
  name: "search",
  label: "",
  placeholder: "Search",
  value: "value",
};

describe("Input", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Input, {
      props: defaultProps,
    });
  });

  it("should have correct name", () => {
    expect(wrapper.find("input").element.name).toBe(defaultProps.name);
  });

  it("should have correct placeholder", () => {
    expect(wrapper.find("input").element.placeholder).toBe(
      defaultProps.placeholder
    );
  });

  it("should not have label if label props is not set", () => {
    expect(wrapper.find("label").exists()).toBe(false);
  });

  it("should have correct label", async () => {
    await wrapper.setProps({ label: "Search" });
    expect(wrapper.find("label").text()).toBe("Search");
  });

  it("snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
