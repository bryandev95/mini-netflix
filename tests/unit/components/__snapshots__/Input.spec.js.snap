// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<div class="input">
  <!--v-if--><input type="text" name="search" placeholder="Search">
</div>
`;
